File: LongestCommonSubsequence.java
/* Longest Common Subsequence (LCS) problem
 * It finds the longest subsequence that is common to two given sequences.
 * In this implementation, the longestCommonSubsequence function takes two strings, text1 and text2, as input. 
 * It returns the length of the longest common subsequence between the two strings.
 */

public class LongestCommonSubsequence {
    public static int longestCommonSubsequence(String text1, String text2) {
        int m = text1.length();
        int n = text2.length();

        int[][] dp = new int[m + 1][n + 1];

        for (int i = 1; i <= m; i++) {
            for (int j = 1; j <=n; j++) {
                if (text1.charAt(i - 1) == text2.charAt(j - 1)) {
                    dp[i][j] = dp[i - 1] + 1;
                } else {
                    dp[i][j] = Math.max(dp[i - 1][j], dp[i][j] - 1]);
                  }
            }
        }
        return dp[m][n];
    }
    public  static void main(String[] args) {
        String text1 = "AGGTAB";
        String text2 = "GXTXAYB";
        int length = longestCommonSubsequence(text1, text2);
        System.out.println("Length of Longest Common Subsequence: " + length);
    }
}

File: Longest Common Subsequence.cpp
#include<bits/stdc++.h>
const int mod=1e9+7;
using namespace std;

 // } Driver Code Ends
// function to find longest common subsequence

class Solution
{
    public:
    //Function to find the length of longest common subsequence in two strings.
       int fun(string text1, string text2,int n,int m, vector<vector<int>> &dp)
    {
        if(dp[n][m]!=-1)return dp[n][m];
        if(n==0 || m==0)return 0;
        
        
        if(text1[n-1] == text2[m-1])
            return dp[n][m] = 1+fun(text1, text2, n-1, m-1,dp);
        else
            return dp[n][m] = max(fun(text1, text2, n, m-1,dp),fun(text1, text2, n-1, m,dp));
    }
    
    int lcs(int n, int m, string text1, string text2)
    {
        
         vector<vector<int>> dp(n+1,vector<int>(m+1,-1));
        return fun(text1,text2,n,m,dp);
    }
};


// { Driver Code Starts.
int main()
{
    int t,n,k,x,y;
    cin>>t;
    while(t--)
    {
        cin>>x>>y;          // Take size of both the strings as input
        string s1,s2;
        cin>>s1>>s2;        // Take both the string as input
        Solution ob;
        cout << ob.lcs(x, y, s1, s2) << endl;
    }
    return 0;
}
  // } Driver Code Ends